### PR TITLE
Process data in models.

### PR DESCRIPTION
**Pull request by fuadmoin**
In this branch I: 

- [ ] created model classes for all tables in the database.
- [ ] set up associations between the model classes.
- [ ] created users, posts, and comments on the rails console.
- [ ] created custom methods on all the model classes.
- [ ] tested the custom methods on the rails console.

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @fuadmoin,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You have some links in the menu headers of the readme that are broken. It is important for professional documentation to not have broken links. The live demo and deployment links are broken. Kindly fix them or remove those links if you have no content for their sections.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_post_counter
+    user.update(posts_counter: user.posts.count)
+  end
+
+  def recent_posts
+    Post.order(created_at: :desc).limit(5)
+  end
```
Comment (omarsalem7) [inline app/models/post.rb]: - [ ] Kindly consider the post model has a method that returns the 5 most recent comments for a given post.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @fuadmoin 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights

- Set up associations between models ✔️
- The Pull Request has a proper title and description. ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+
+  def update_post_counter
+    user.update(posts_counter: user.posts.count)
+  end
```
Comment (Abenezer-Tilahun) [inline app/models/post.rb]: - [ Optional ] Kindly I recommend you  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_post_counter method every time a new post has been saved to database, then make the update_post_counter method private Just like this :- 

  # app/models/post.rb
```
  after_save :update_post_counter

  private

  def update_post_counter
        user.update(posts_counter: user.posts.count)
  end
```

File: app/models/post.rb (hunk 1 above)
Comment (Abenezer-Tilahun) [reply app/models/post.rb]: Fixed.

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  def update_comment_counter
+    post.update(comments_counter: post.comments.count)
+  end
```
Comment (Abenezer-Tilahun) [inline app/models/comment.rb:11]: - [ Optional ] Kindly I recommend you  use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #update_comments_counter method every time a new comment has been saved to database, then make the update_comments_counter method private Just like this :- 

  # app/models/comment.rb
```
  after_save :update_comments_counter

  private

  def update_comments_counter
       post.update(comments_counter: post.comments.count)
  end
```

**Review by Abenezer-Tilahun — APPROVED** [review]
Hi @fuadmoin 
 
Your project is complete  👍 💯   There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 👏🏻

✔️ No linter errors.
✔️ You used correct Git-Flow.
✔️ Professional README  
✔️ Good Commit Messages
✔️ Good job for Implemented Suggested changes.

## Optional suggestions 
 
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

**Pull request merged**